config/CLIConfiguration.ts

```diff
@@ -204,16 +204,21 @@ class _CLIConfiguration {
       return null;
     }
 
-    if (typeof nameOrIdToCheck === 'number') {
-      accountId = nameOrIdToCheck;
-    } else if (/^\d+$/.test(nameOrIdToCheck)) {
-      accountId = parseInt(nameOrIdToCheck, 10);
-    } else {
+    if (typeof nameOrIdToCheck === 'string') {
       name = nameOrIdToCheck;
+      if (/^\d+$/.test(nameOrIdToCheck)) {
+        accountId = parseInt(nameOrIdToCheck, 10);
+      }
```

Comment:
If the 'nameOrIdToCheck` is a string **and** numeric, we set both `name` and `accountId` to that value. 

Then, we check whether it is present in `this.config.accounts` using both the `name` and `accountId` property. 

This allows us to account for two scenarios:
1) A customer names an account something numeric (ie. "2024", "1956" etc).
2) A customer supplies an account ID to a command (`hs account use 12345678`), which the CLI interprets it as a string.